### DISCUSSION TITLE
Java API cannot start fdtransfer?

### DISCUSSION THREAD

**Discussion by Winson-Huang** [async-profiler/async-profiler#796] [Q&A] [answered]
By now, is there a way to start fdtransfer using Java API? Thanks for answering!

**Comment by Jongy** [comment, marked as answer]
The idea behind fdtransfer is that the Java process itself lacks the necessary permissions to enable profiling with perf_events, so an external program (which has the necessary privileges) uses them and grants the perf_events "handle" to the profiled Java process.

If your Java process has the privileges then you don't need fdtransfer at all :)
If it doesn't have them, you can't enable profiling from the Java application itself - it needs to receive the perf_events handle from a more privileged process.
